2023.1 cherries

mnasiadka opened the pull request.

jovial commented on ansible/roles/ipa-images/tasks/set-driver-info.yml:
```diff
@@ -32,7 +32,7 @@
     - item.Name is not none
     - item.Name not in groups["baremetal-compute"]
-  with_items: "{{ ipa_images_ironic_node_list.stdout | from_json }}"
+  with_items: "{{ ipa_images_ironic_node_list.stdout | default('{}') | from_json }}"
```
I guess we this upstream aswell. Although I'm surprised that it gets evaluated given the `when` condition on the include. Is there a difference in the behaviour of `when` between `include` and `include_tasks`?

markgoddard replied:
I'm not sure about include, but include_tasks will completely ignore the tasks in the file if the condition is false. Include might be more like import_tasks, which is a static include.

dougszumski committed:
Add support for caso

(cherry picked from commit a78a8fdd3b7653d37effb2088bc8aa6e5ce23eaf)
(cherry picked from commit 5f57c8c9ac5c0d6a8cfe4a28fd30a5638eb61aa1)

Change-Id: I8804bd4647140bc837832a0cb6d8bbf3f8e53ac8
(cherry picked from commit 609f822c3f92625d56cdfff3aed2f793ad569fa6)
(cherry picked from commit 6862b1dd293c58f68b780c26b538a675244481f8)
(cherry picked from commit 2e9b8dfc3a3cf2c9b72666cbb33fa4507bfdb6c1)
(cherry picked from commit 219ed0bc2df928f731858deca0ea67383fafc8e0)
(cherry picked from commit f0258514307fcd94c9fb9aab9088f60f264fe0f3)
(cherry picked from commit fbe8ba4d5daf6ab17866a9000a5eaa24d1f05145)

markgoddard committed:
Fix firewalld configuration for monitoring hosts

There was a stray double quote in the firewalld config for monitoring
hosts, resulting in the following error when firewalld is enabled:

  ValueError: No closing quotation

Closes-Bug: 2019867
Change-Id: I201faf14da5d143670250052ab15fc285f24868c
(cherry picked from commit b7e95c83d3ea22bdccb5a5440e677c8a36f4cf61)

priteau committed:
Speed up calls to Bifrost

For overcloud commands involving Bifrost (e.g. overcloud provision), we
use the Bifrost dynamic inventory. This runs bifrost_inventory.py which
gathers node and port information for all bare metal hosts. Because this
gets executed for each node in our inventory, it produces a large number
of API requests to Ironic, which grows as the number of hosts increases.

Halve the number of calls using the BIFROST_NODE_NAMES environment
variable, which restricts fetching port information to just the node
being queried. bifrost_inventory.py still performs one call for each
node, which must be fixed separately in Bifrost [1].

Time to query overcloud nodes' hardware introspection data for 44 hosts:

* before patch: 823 seconds
* after patch: 415 seconds

With the patch to Bifrost [1], this is further reduced to 46 seconds.

[1] https://review.opendev.org/c/openstack/bifrost/+/882950

Change-Id: I341075115442a38c327e3ade74b81b1b9e7e85c6
(cherry picked from commit 3e03884c476604a717c1fa5fb35771f60e39dd8f)

Commit:
Merge "Speed up calls to Bifrost" into stable/2023.1
